### PR TITLE
Sizes issue 352

### PR DESCRIPTION
**Pull request by shrutir25**
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/466)

### CODE REVIEW

**Comment by shrutir25** [comment]
@anhowe - could you take a look ?

File: parts/dcosagentresourcesvmss.t [hunk 1]
```diff
@@ -189,7 +189,7 @@
       "sku": {
         "capacity": "[variables('{{.Name}}Count')]",
         "name": "[variables('{{.Name}}VMSize')]",
-        "tier": "Standard"
+        "tier": "[variables('{{.Name}}VMSizeTier')]"
```
Comment (colemickens) [inline parts/dcosagentresourcesvmss.t]: Is this change still needed if we're not supporting 'Basic'?

Comment (colemickens) [inline parts/dcosagentresourcesvmss.t]: Or is this so that everything still works even if the customer manually tries to edit and use a 'Basic_X' vm size?

Comment (anhowe) [inline parts/dcosagentresourcesvmss.t]: Even though we don't use basic, there may be new sizes in the future, and the error produced is hard to troubleshoot, so we should fix it.

File: pkg/acsengine/Get-AzureConstants.ps1 [hunk 2]
```diff
@@ -27,7 +30,7 @@ Get-AllSizes() {
 		#Write-Output $location.Location
 		$sizes = Get-AzureRmVMSize -Location $location.Location
 		ForEach ($size in $sizes) {
-			if (!$sizeMap.ContainsKey($size.Name)) {
+			if (!$sizeMap.ContainsKey($size.Name) -and !($size.Name.split('_')[0] -eq 'BASIC')) {
```
Comment (anhowe) [inline pkg/acsengine/Get-AzureConstants.ps1]: 1. Consider case insensitive comparison
2. add a comment to why basic is being filtered out (no LB support)

File: parts/dcosagentresourcesvmss.t (hunk 1 above)
Comment (colemickens) [inline parts/dcosagentresourcesvmss.t]: 👍 thanks!

File: pkg/acsengine/Get-AzureConstants.ps1 (hunk 2 above)
Comment (shrutir25) [inline pkg/acsengine/Get-AzureConstants.ps1]: As discussed, sizes are only in 'Basic' format, so updated the if condition accordingly.

**Comment by shrutir25** [comment]
@anhowe  I have made the changes. Could you take a look ?

**Comment by anhowe** [comment]
I just noticed one thing, @shrutir25 please scrub examples and tests to make sure no basic is listed.

**Comment by JackQuincy** [comment]
most recent change looks good. I'll merge this.

**Comment by dmitsh** [comment]
Did we run Jenkins PR validation job for this PR?

**Comment by JackQuincy** [comment]
Apparently not my bad. I will send out a PR to get us back in a good state

**Comment by dmitsh** [comment]
It is rather my bad :) I thought we configured GitHub not to allow PRs without building them first ...
